From ccd675b492845eccb2d8d8135281713936aab26c Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Sat, 28 Dec 2019 02:13:55 +0100
Subject: [PATCH 68/98] ARM: dts: mxiii: add support for the audio outputs -
 WiP

The MXIII and MXIII-Plus boards come with SPDIF and I2S Stereo DAC
(provided by a Beijing Times Minxin Technology Co., Ltd. MXT8234) audio
outputs. The latter is not supported yet (as we are missing a driver for
it). Also the I2S output can be routed to the HDMI TX controller.

Add the SPDIF output as well as HDMI TX audio routing on these boards.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 arch/arm/boot/dts/amlogic/meson8m2-mxiii.dtsi | 62 +++++++++++++++++++
 1 file changed, 62 insertions(+)

diff --git a/arch/arm/boot/dts/amlogic/meson8m2-mxiii.dtsi b/arch/arm/boot/dts/amlogic/meson8m2-mxiii.dtsi
index c25a9f7da3a08b..ee61e2ebcf657e 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2-mxiii.dtsi
+++ b/arch/arm/boot/dts/amlogic/meson8m2-mxiii.dtsi
@@ -69,6 +69,59 @@
 		clock-names = "ext_clock";
 	};
 
+	sound {
+		compatible = "amlogic,gx-sound-card";
+		model = "TRONSMART-MXIII";
+
+		clocks = <&clkc CLKID_MPLL0>,
+			 <&clkc CLKID_MPLL1>;
+
+		assigned-clocks = <&clkc CLKID_MPLL0>,
+				  <&clkc CLKID_MPLL1>;
+		assigned-clock-rates = <270950400>,
+				       <294912000>;
+
+		dai-link-0 {
+			sound-dai = <&aiu AIU_CPU CPU_I2S_FIFO>;
+		};
+
+		dai-link-1 {
+			sound-dai = <&aiu AIU_CPU CPU_SPDIF_FIFO>;
+		};
+
+		dai-link-2 {
+			sound-dai = <&aiu AIU_CPU CPU_I2S_ENCODER>;
+			dai-format = "i2s";
+			mclk-fs = <256>;
+
+			codec-0 {
+				sound-dai = <&aiu AIU_HDMI CTRL_I2S>;
+			};
+		};
+
+		dai-link-3 {
+			sound-dai = <&aiu AIU_HDMI CTRL_OUT>;
+
+			codec-0 {
+				sound-dai = <&hdmi_tx 0>;
+			};
+		};
+
+		dai-link-4 {
+			sound-dai = <&aiu AIU_CPU CPU_SPDIF_ENCODER>;
+
+			codec-0 {
+				sound-dai = <&spdif_dit>;
+			};
+		};
+	};
+
+	spdif_dit: spdif-audio-codec {
+		compatible = "linux,spdif-dit";
+		#sound-dai-cells = <0>;
+		sound-name-prefix = "DIT";
+	};
+
 	vcc_3v3: regulator-vcc3v3 {
 		compatible = "regulator-fixed";
 		regulator-name = "VCC3V3";
@@ -84,6 +137,15 @@
 	};
 };
 
+&aiu {
+	status = "okay";
+
+	pinctrl-0 = <&i2s_am_clk_pins>, <&i2s_out_ao_clk_pins>,
+		    <&i2s_out_lr_clk_pins>, <&i2s_out_ch01_ao_pins>,
+		    <&spdif_out_pins>;
+	pinctrl-names = "default";
+};
+
 &cpu0 {
 	cpu-supply = <&vcck>;
 };

